custom/enikshay/reports/choice_providers.py

```diff
@@ -0,0 +1,19 @@
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.userreports.reports.filters.choice_providers import LocationChoiceProvider
+
+
+class DistrictChoiceProvider(LocationChoiceProvider):
+    def _locations_query(self, query_text, user):
+        active_locations = SQLLocation.active_objects
+        if query_text:
+            return active_locations.filter_path_by_user_input(
+                domain=self.domain,
+                location_type__code__exact='dto',
```

Comment:
I don't think this is a valid parameter for `filter_path_by_user_input` - you may just want `active_locations.filter_by_user_input(...).filter(location_type__code__exact='dto')`
(assuming that what you want is for users to type only dto locations)

Reply:
Actually, this is being filtered again later anyways, so you can probably just drop this line.  Switch to `filter_by_user_input` if what you want is for typing to match against `dto` names only, otherwise this will match against `dto`s or their parents.